pkg/cloudclient/aws/private.go

```diff
@@ -129,8 +129,7 @@ func updateAWSLBList(kclient client.Client, oldLBList []awsproviderapi.LoadBalan
 			return err
 		}
 		machineToPatch.Spec.ProviderSpec = *newProviderSpecEncoded
-		machineObj := machineToPatch.DeepCopyObject()
```

Comment:
the `machineToPatch` itself is being passed to the patch on the next line.
```go
if err := kclient.Patch(context.Background(), &machineToPatch, baseToPatch); err != nil {
```
Why would we need machineObj being created? 

additionally, the version below doesn't compile
```go
machineToPatch.Spec.ProviderSpec = *newProviderSpecEncoded
machineObj := machineToPatch.DeepCopyObject()
if err := kclient.Patch(context.Background(), machineObj, baseToPatch); err != nil {
	log.Error(err, "Failed to update LBs in machine's providerSpec", "machine", machineToPatch.Name)
	return err
}
```
due to `cannot use machineObj (variable of type runtime.Object) as client.Object value in argument to kclient.Patch: missing method GetAnnotations compilerInvalidIfaceAssign`

Reply:
Ah -- because controller-runtime changed the signature of `Patch`.

> Why would we need machineObj being created?

In theory, if `machineObj` were being passed to this method by reference, or if by-value did a shallow copy, this

```go
machineToPatch.Spec.ProviderSpec = *newProviderSpecEncoded
```

would affect the object in the calling scope, which could have unintended side effects.

Reply:
ahh, got your concern now, I've changed it to be using DeepCopy()